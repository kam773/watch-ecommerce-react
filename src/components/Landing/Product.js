import React from "react";
import styled from "styled-components";
import { ProductConsumer } from "../../context";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import { Card } from "reactstrap";

export default class Product extends React.Component {
  render() {
    const { id, title, img, price } = this.props.product;
    return (
      <ProductWrapper className="col-9 mx-auto col-md-6 col-lg-3 my-3">
        <Card>
          <ProductConsumer>
            {value => (
              <div
                className="img-container p-5"
                onClick={() => value.handleDetail(id)}
              >
                <Link to="/details">
                  <img src={img} alt="" className="card-img-top" />
                </Link>
              </div>
            )}
          </ProductConsumer>
          {/* Card footer */}
          <div className="card-footer d-flex justify-content-between">
            <p className="align-self-center mb-0"> {title} </p>
            <h5 className="text-blue font-italic mb-0">
              <span className="mr-1"> $ {price} </span>
            </h5>
          </div>
        </Card>
      </ProductWrapper>
    );
  }
}

Product.propTypes = {
  product: PropTypes.shape({
    id: PropTypes.number,
    img: PropTypes.string,
    title: PropTypes.string,
    price: PropTypes.number,
    inCart: PropTypes.bool
  }).isRequired
};

const ProductWrapper = styled.div`
  .cart-btn {
    background: transparent;
    border: none;
    cursor: pointer;
    padding: 0.8rem;
    font-size: 1.5rem;
    position: absolute;
    bottom: 0;
    right: 0;
  }
  .img-container {
    position: relative;
    overflow: hidden;
  }
  .card-img-top {
    transition: all 0.5s linear;
  }
  .img-container:hover .card-img-top {
    transform: scale(1.2);
  }
`;
