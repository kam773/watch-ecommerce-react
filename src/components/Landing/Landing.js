import React from "react";
import MySlider from "../Slider/Slider";
import styled from "styled-components";
import Banner1 from "../../img/abt-1.jpg";
import Banner2 from "../../img/abt-2.jpg";
import Banner3 from "../../img/abt-3.jpg";
import { Container, Row, Col } from "reactstrap";
import Product from "./Product";
import { ProductConsumer } from "../../context";

export default function Landing() {
  return (
    <SliderContainer>
      <MySlider />
      <Container className="py-5">
        <Row>
          <Col md="4">
            <figure className="img-wrapper hover-effect">
              <img className="img-responsive" src={Banner1} alt="Banner" />
              <figcaption>
                <h4>Nulla maximus nunc</h4>
                <p>In sit amet sapien eros Integer dolore magna aliqua</p>
              </figcaption>
            </figure>
          </Col>
          <Col md="4">
            <figure className="img-wrapper hover-effect">
              <img className="img-responsive" src={Banner2} alt="Banner" />
              <figcaption>
                <h4>Mauris erat augue</h4>
                <p>In sit amet sapien eros Integer dolore magna aliqua</p>
              </figcaption>
            </figure>
          </Col>
          <Col md="4">
            <figure className="img-wrapper hover-effect">
              <img className="img-responsive" src={Banner3} alt="Banner" />
              <figcaption>
                <h4>Cras elit mauris</h4>
                <p>In sit amet sapien eros Integer dolore magna aliqua</p>
              </figcaption>
            </figure>
          </Col>
        </Row>
        <h1 className="text-center text-black text-capitalize my-3">
          featured products
        </h1>
        <Row>
          <ProductConsumer>
            {value => {
              return value.featured.map(product => {
                return <Product key={product.id} product={product} />;
              });
            }}
          </ProductConsumer>
        </Row>
        <h1 className="text-center text-black text-capitalize my-3">
          Best Sellers
        </h1>
        <Row>
          <ProductConsumer>
            {value => {
              return value.featured.map(product => {
                return <Product key={product.id} product={product} />;
              });
            }}
          </ProductConsumer>
        </Row>
      </Container>
    </SliderContainer>
  );
}

const SliderContainer = styled.div`
  width: 100%;
  height: auto;

  .img-wrapper {
    position: relative;
  }
  figcaption {
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    height: 100%;
    width: 100%;
    opacity: 0;
    transition: 0.5s ease;
    background-color: #008cba;
    width: 100%;
  }
  .img-wrapper:hover figcaption {
    opacity: 0.8;
  }
  .img-wrapper:hover figcaption h4 {
    opacity: 1;
  }
  .img-wrapper:hover figcaption p {
    opacity: 1;
  }
  figcaption h4 {
    color: #fff;
    font-size: 1.4rem;
    position: absolute;
    top: 30%;
    left: 50%;
    transform: translate(-50%, -50%);
    text-transform: capitalize;
    letter-spacing: 2px;
    text-align: center;
    opacity: 1;
  }
  figcaption p {
    color: white;
    font-size: 1.2rem;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    text-align: center;
    opacity: 1;
    margin-top: 30px;
    line-height: 1.4;
  }
  .cart-btn {
    background: transparent;
    border: none;
    cursor: pointer;
    text-align: right;
    padding: 0.8rem;
    font-size: 1.5rem;
  }
  .img-container {
    position: relative;
    overflow: hidden;
  }
  .card-img-top {
    transition: all 0.5s linear;
  }
  .img-container:hover .card-img-top {
    transform: scale(1.2);
  }
`;
