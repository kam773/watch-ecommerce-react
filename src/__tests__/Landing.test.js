import React from 'react';
import Landing from '../components/Landing/Landing';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';

describe("Landing", () => {
    it("renders without crashes", () => {
        const tree = renderer.create(<Landing/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("has figure tag", () => {
        const wrapper = shallow(<Landing />);
        expect(wrapper.find('figure').exists()).toEqual(true);
    });
    it("has figcaption tag", () => {
        const wrapper = shallow(<Landing />);
        expect(wrapper.find('figcaption').exists()).toEqual(true);
    });
    it("has h1 tag", () => {
        const wrapper = shallow(<Landing />);
        expect(wrapper.find('h1').exists()).toEqual(true);
    });
    it("has p tag", () => {
        const wrapper = shallow(<Landing />);
        expect(wrapper.find('p').exists()).toEqual(true);
    });
    it("has a title", () => {
        const wrapper = mount(<h1>Featured Products</h1>)
        expect(wrapper.text()).toEqual('Featured Products');
    });
    it("renders heading tag", () => {
        const wrapper = shallow(<Landing/>);
        expect(wrapper.find('h1')).toHaveLength(2);
    });
    it("has img tag", () => {
        const wrapper = shallow(<Landing />);
        expect(wrapper.find('img').exists()).toEqual(true);
    });
    it("has class card", () => {
        const wrapper = mount(<Landing />);
        expect(wrapper.find('.card').every('.card')).toEqual(true);
    });
    it("has class card-img-top", () => {
        const wrapper = mount(<Landing />);
        expect(wrapper.find('.card-img-top').every('.card-img-top')).toEqual(true);
    });
    it("has button to have class cart-btn", () => {
        const wrapper = mount(<Landing />);
        expect(wrapper.find('.cart-btn').every('.cart-btn')).toEqual(true);
    });
    it("has a cart icon", () => {
        const wrapper = mount(<Landing />);
        expect(wrapper.find('i').exists()).toEqual(true);
    });
    it("has button to have class cart-btn", () => {
        const wrapper = mount(<Landing />);
        expect(wrapper.find('.cart-btn').every('.cart-btn')).toEqual(true);
    });


})