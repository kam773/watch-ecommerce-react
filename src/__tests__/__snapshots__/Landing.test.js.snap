// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing renders without crashes 1`] = `
<div
  className="sc-bdVaJa dYVmkF"
>
  <div
    className="slider-container"
  >
    <div
      className="slick-slider slick-initialized"
      dir="ltr"
    >
      <div
        className="slick-list"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={null}
        onMouseMove={null}
        onMouseUp={[Function]}
        onTouchCancel={null}
        onTouchEnd={[Function]}
        onTouchMove={null}
        onTouchStart={[Function]}
        style={Object {}}
      >
        <div
          className="slick-track"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <div
            aria-hidden={false}
            className="slick-slide slick-active slick-current"
            data-index={0}
            onClick={[Function]}
            style={
              Object {
                "WebkitTransition": "opacity 500ms linear, visibility 500ms linear",
                "left": -0,
                "opacity": 1,
                "outline": "none",
                "position": "relative",
                "transition": "opacity 500ms linear, visibility 500ms linear",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="img-wrapper"
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt="Banner"
                  src="bnr-1.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="slick-slide"
            data-index={1}
            onClick={[Function]}
            style={
              Object {
                "WebkitTransition": "opacity 500ms linear, visibility 500ms linear",
                "left": -0,
                "opacity": 0,
                "outline": "none",
                "position": "relative",
                "transition": "opacity 500ms linear, visibility 500ms linear",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="img-wrapper"
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt="Banner"
                  src="bnr-2.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="slick-slide"
            data-index={2}
            onClick={[Function]}
            style={
              Object {
                "WebkitTransition": "opacity 500ms linear, visibility 500ms linear",
                "left": -0,
                "opacity": 0,
                "outline": "none",
                "position": "relative",
                "transition": "opacity 500ms linear, visibility 500ms linear",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="img-wrapper"
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt="Banner"
                  src="bnr-3.jpg"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <ul
        className="slick-dots"
        onMouseEnter={null}
        onMouseLeave={null}
        onMouseOver={null}
        style={
          Object {
            "display": "block",
          }
        }
      >
        <li
          className="slick-active"
        >
          <button
            onClick={[Function]}
          >
            1
          </button>
        </li>
        <li
          className=""
        >
          <button
            onClick={[Function]}
          >
            2
          </button>
        </li>
        <li
          className=""
        >
          <button
            onClick={[Function]}
          >
            3
          </button>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="py-5"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-4"
        >
          <figure
            className="img-wrapper hover-effect"
          >
            <img
              alt="Banner"
              className="img-responsive"
              src="abt-1.jpg"
            />
            <figcaption>
              <h4>
                Nulla maximus nunc
              </h4>
              <p>
                In sit amet sapien eros Integer dolore magna aliqua
              </p>
            </figcaption>
          </figure>
        </div>
        <div
          className="col-md-4"
        >
          <figure
            className="img-wrapper hover-effect"
          >
            <img
              alt="Banner"
              className="img-responsive"
              src="abt-2.jpg"
            />
            <figcaption>
              <h4>
                Mauris erat augue
              </h4>
              <p>
                In sit amet sapien eros Integer dolore magna aliqua
              </p>
            </figcaption>
          </figure>
        </div>
        <div
          className="col-md-4"
        >
          <figure
            className="img-wrapper hover-effect"
          >
            <img
              alt="Banner"
              className="img-responsive"
              src="abt-3.jpg"
            />
            <figcaption>
              <h4>
                Cras elit mauris
              </h4>
              <p>
                In sit amet sapien eros Integer dolore magna aliqua
              </p>
            </figcaption>
          </figure>
        </div>
      </div>
      <h1
        className="text-center text-black text-capitalize my-3"
      >
        featured products
      </h1>
      <div
        className="row"
      >
        <div
          className="col-md-3"
        >
          <div
            className="card"
          >
            <div
              className="img-container p-5"
            >
              <img
                alt="Product"
                className="card-img-top"
                src="p-4.png"
              />
            </div>
            <button
              className="cart-btn"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
            <div
              className="card-footer d-flex justify-content-between"
            >
              <p
                className="align-self-center mb-0"
              >
                CASSIO Classic
              </p>
              <h5
                className="text-blue font-italic mb-0"
              >
                <span
                  className="mr-1"
                >
                   $ 89 
                </span>
              </h5>
            </div>
          </div>
        </div>
        <div
          className="col-md-3"
        >
          <div
            className="card"
          >
            <div
              className="img-container p-5"
            >
              <img
                alt="Product"
                className="card-img-top"
                src="p-2.png"
              />
            </div>
            <button
              className="cart-btn"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
            <div
              className="card-footer d-flex justify-content-between"
            >
              <p
                className="align-self-center mb-0"
              >
                ATLANTIC sealine
              </p>
              <h5
                className="text-blue font-italic mb-0"
              >
                <span
                  className="mr-1"
                >
                   $ 199 
                </span>
              </h5>
            </div>
          </div>
        </div>
        <div
          className="col-md-3"
        >
          <div
            className="card"
          >
            <div
              className="img-container p-5"
            >
              <img
                alt="Product"
                className="card-img-top"
                src="p-6.png"
              />
            </div>
            <button
              className="cart-btn"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
            <div
              className="card-footer d-flex justify-content-between"
            >
              <p
                className="align-self-center mb-0"
              >
                FESTINA Multifunction
              </p>
              <h5
                className="text-blue font-italic mb-0"
              >
                <span
                  className="mr-1"
                >
                   $ 169 
                </span>
              </h5>
            </div>
          </div>
        </div>
        <div
          className="col-md-3"
        >
          <div
            className="card"
          >
            <div
              className="img-container p-5"
            >
              <img
                alt="Product"
                className="card-img-top"
                src="p-7.png"
              />
            </div>
            <button
              className="cart-btn"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
            <div
              className="card-footer d-flex justify-content-between"
            >
              <p
                className="align-self-center mb-0"
              >
                SEIKO Prospex
              </p>
              <h5
                className="text-blue font-italic mb-0"
              >
                <span
                  className="mr-1"
                >
                   $ 189 
                </span>
              </h5>
            </div>
          </div>
        </div>
      </div>
      <h1
        className="text-center text-black text-capitalize my-3"
      >
        Best Sellers
      </h1>
      <div
        className="row"
      >
        <div
          className="col-md-3"
        >
          <div
            className="card"
          >
            <div
              className="img-container p-5"
            >
              <img
                alt="Product"
                className="card-img-top"
                src="p-2.png"
              />
            </div>
            <button
              className="cart-btn"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
            <div
              className="card-footer d-flex justify-content-between"
            >
              <p
                className="align-self-center mb-0"
              >
                FESTINA Multifunction
              </p>
              <h5
                className="text-blue font-italic mb-0"
              >
                <span
                  className="mr-1"
                >
                   $ 169 
                </span>
              </h5>
            </div>
          </div>
        </div>
        <div
          className="col-md-3"
        >
          <div
            className="card"
          >
            <div
              className="img-container p-5"
            >
              <img
                alt="Product"
                className="card-img-top"
                src="p-7.png"
              />
            </div>
            <button
              className="cart-btn"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
            <div
              className="card-footer d-flex justify-content-between"
            >
              <p
                className="align-self-center mb-0"
              >
                SEIKO Prospex
              </p>
              <h5
                className="text-blue font-italic mb-0"
              >
                <span
                  className="mr-1"
                >
                   $ 189 
                </span>
              </h5>
            </div>
          </div>
        </div>
        <div
          className="col-md-3"
        >
          <div
            className="card"
          >
            <div
              className="img-container p-5"
            >
              <img
                alt="Product"
                className="card-img-top"
                src="p-4.png"
              />
            </div>
            <button
              className="cart-btn"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
            <div
              className="card-footer d-flex justify-content-between"
            >
              <p
                className="align-self-center mb-0"
              >
                CASSIO Classic
              </p>
              <h5
                className="text-blue font-italic mb-0"
              >
                <span
                  className="mr-1"
                >
                   $ 169 
                </span>
              </h5>
            </div>
          </div>
        </div>
        <div
          className="col-md-3"
        >
          <div
            className="card"
          >
            <div
              className="img-container p-5"
            >
              <img
                alt="Product"
                className="card-img-top"
                src="p-6.png"
              />
            </div>
            <button
              className="cart-btn"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
            <div
              className="card-footer d-flex justify-content-between"
            >
              <p
                className="align-self-center mb-0"
              >
                ATLANTIC sealine
              </p>
              <h5
                className="text-blue font-italic mb-0"
              >
                <span
                  className="mr-1"
                >
                   $ 199 
                </span>
              </h5>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
